k8s/demo/crunchy-postgres/README.md

```diff
@@ -1,35 +1,203 @@
+- Verify table creation is restricted on the replica
+
+### Step-1: Install the PostgreSQL-Client
+
+Install the PostgreSQL CLient Utility (psql) on any of the Kubernetes machines to perform database operations 
+from command line. 
+
+```
+sudo apt-get install postgresql-client
+```
+
+### Step-2: Verify Cluster Replication Status on Crunchy-Postgres Cluster
+
+Identify the IP address of the primary (pgset-0) pod or the service (pgset-primary) and execute the following
+query:
+
+```
+test@Master:~$ psql -h 10.47.0.3 -U testuser postgres -c 'select * from pg_stat_replication'
+
+ pid | usesysid |   usename   | application_name | client_addr | client_hostname | client_port |         backend_start         | backend_xmin |   state   | sent_lsn  | write_lsn | flush_lsn | replay_lsn | write_lag | flush_lag | replay_lag | sync_priority | sync_state
+-----+----------+-------------+------------------+-------------+-----------------+-------------+-------------------------------+--------------+-----------+-----------+-----------+-----------+------------+-----------+-----------+------------+---------------+------------
+  94 |    16391 | primaryuser | pgset-1          | 10.44.0.0   |                 |       60460 | 2017-11-14 09:29:21.990782-05 |              | streaming | 0/3014278 | 0/3014278 | 0/3014278 | 0/3014278  |           |           |            |             0 | async
+(1 row)
 ```
 
-### Step 2: Run the StatefulSet. 
-The files are make available with default images and credentails (set.json)
+The replica should be registered for "asynchronous" replication. 
+
+### Step-3: Create a Table with Test Content on the Default Database 
+
+These queries should be executed on the primary.
+
 ```
-./run.sh
+test@Master:~$ psql -h 10.47.0.3 -U testuser postgres -c 'create table foo(id int)'
+Password for user testuser:
+CREATE TABLE
 ```
+```
+test@Master:~/crunchy-postgres$ psql -h 10.47.0.3 -U testuser postgres -c 'insert into foo values (1)'
+Password for user testuser:
+INSERT 0 1
+```
+
+### Step-4: Verify Data Synchronization on Replica
+
+Identify the IP address of the replica (pgset-1) pod or the service (pgset-replica) and execute the following
+query: 
 
-The above step will automatically create the OpenEBS volumes required for master and replica postgresql containers. The volume details can be inspected using the standard kubectl commands like:
 ```
-kubectl get pvc
-kubectl get pv
+test@Master:~$ psql -h 10.44.0.6 -U testuser postgres -c 'table foo'
+Password for user testuser:
+ id
+----
+  1
+(1 row)
 ```
 
+Verify that the table content is replicated successfully
 
-## Motivation
+### Step-5: Verify Database Write Restricted on Replica
 
-The k8s spec files are based on the files provided by [CrunchyData StatefulSet with Dynamic Provisioner](https://github.com/CrunchyData/crunchy-containers/tree/master/examples/kube/statefulset-dyn)
+Attempt creation of a new table on the replica and verify the operation is unsuccessful.
```

Comment:
Attempt to create a new table on the replica, and verify that the table creation is unsuccessful.